generic_config_updater/patch_sorter.py

```diff
@@ -333,6 +333,114 @@ def _extend_moves(self, move, diff):
             for newmove in extender.extend(move, diff):
                 yield newmove
 
+class ConfigFilter:
```

Comment:
>ConfigFilter [](http://example.com/codeflow?start=6&length=12)

The class is more relating to JsonPath. It's a general concept instead of specific to config.

Reply:
I think the way to filter is implementation detail to this class, they are passed to the constructor. This is because the class expose a single function to `get_paths` and paths are JsonPointer.

If we are going to use JSONPath, will use it is instead of the current `Patterns`. This is be

Can I have a follow-up on this later as I will have to:
- JSONpath can return list of JSONPaths which I have to convert to JsonPointer Paths (in current impl it is the function `get_paths`)
- Look for how to add a parameter to the JsonPath expression (in current impl it is the `@` symbol)

I think in general I will need this class `ConfigFilter`.

Reply:
This comment is about the name. How about JsonPathFilter, or something similar.

Reply:
Renamed to JsoinPointerFilter